src/instrumentation/podmutator.go

```diff
@@ -246,7 +246,7 @@ func (pm *instPodMutator) MutateWithAnyAvailInstrumentations(ctx context.Context
 		// we still allow the pod to be created, but we log a message to the operator's logs
 		logger.Error(err, "failed to select a New Relic Instrumentation instance for this pod")
 		return pod, err
-	} else if instCandidates == nil {
+	} else if len(instCandidates) == 0 {
```

Comment:
Thanks for catching this! I had meant to flag this one.

Reply:
Can we remove these two lines
```
	var instCandidates []*v1alpha1.Instrumentation
	var err error
```
and 
use
`if instCandidates, err := ...{
} 
if len(instCandidates) == 0 {
....
`

Reply:
updated